proposals/interface-types/working-notes/Instructions.md

```diff
@@ -0,0 +1,252 @@
+# Interface instructions
+
+This proposal defines new instructions that operate over interface value types. These instructions have the same form as core instructions; they push and pop values on an execution stack and support single pass validation.
+
+The new interface instructions are summarized below.
+
+```
+#interface-instr ::=
+  | bool.lift_iMM
+  | bool.lower_iMM
+
+  | sNN.lift_iMM
+  | sNN.lower_iMM
+
+  | uNN.lift_iMM
+  | uNN.lower_iMM
+
+  | string.lift_memory ..
+  | string.lower_memory ..
+
+  | record.lift ..
+  | record.lower ..
+
+  | array.lift_memory ..
+  | array.lower_memory ..
```

Comment:
This is likely not a single instruction.
You need to allocate, (not a separate instruction), you need to 'stride' through the array, you also need an

array.size
or 
array.count
or 
both

Reply:
I've fused the allocate/'stride' instructions together as is done in the 'Explainer' for strings. What are 'array.size/count' needed for?

Reply:
see above

Reply:
BTW, the explainer is probably a bit out of date; especially wrt handling arrays and records.

Reply:
I've separated memory allocation from lowering instructions and added `array.count` and `string.size`.